nano/node/wallet.cpp

```diff
@@ -1512,6 +1550,107 @@ void nano::wallets::reload ()
 	}
 }
 
+void nano::wallets::do_work_regeneration ()
+{
+	std::unique_lock<std::mutex> regeneration_lock (difficulty_mutex);
+	while (!stopped)
+	{
+		if (!difficulty_reque.empty ())
+		{
+			auto const first (difficulty_reque.begin ());
+			auto now (std::chrono::steady_clock::now ());
+			auto queued (first->first);
+			auto block (first->second);
+			std::shared_ptr<nano::block> block_l (block);
+			auto is_confirmed = [this, block]() -> bool {
+				bool confirmed_l (false);
+				std::unique_lock<std::mutex> lock (this->node.active.mutex);
+				auto existing (this->node.active.roots.find (block->qualified_root ()));
+				if (this->node.active.roots.end () != existing)
```

Comment:
Could these three lines (including the lock) be replaced with a call to `this->node.active.active (*block)` ? To avoid using the mutex directly.